hack/update-bazel.sh

```diff
@@ -19,6 +19,10 @@ set -o pipefail
 
 cd "$(git rev-parse --show-toplevel)"
 mkdir -p ./vendor
-touch ./vendor/BUILD.bazel
+if [[ ! -e ./vendor/BUILD.bazel ]]; then
+  echo "Bootstrapping vendor..." >&2
+  touch ./vendor/BUILD.bazel
+  bazel run //:gazelle-bootstrap
```

Comment:
Why do we do this only when `./vendor/BUILD.bazel` does not exist? What if `./vendor` is updated in a way that necessitates a change to the BUILD files?

Reply:
If someone updates something under in vendor, then the `bazel run //:gazelle` command will fix everything in vendor up.

If is extremely odd if `./vendor/BUILD.bazel` doesn't exist. This most likely means that we just recreated the vendor dir during an `update-deps.sh` which will `rm -rf vendor && go mod vendor`. At this time, there are no BUILD.bazel files, so we cannot use the vendored version of gazelle. Instead we do an initial pass with the (potentially wrong) version included with the bazel-gazelle WORKSPACE repo rule. This creates the vendored gazelle targets.

Right after this we run `bazel run //:gazelle` which will use the vendored version of gazelle to ensure we're generating things correctly.

Normally we don't delete vendor before running `hack/update-bazel.sh` so vendored gazelle rule already exists so we don't need to bootstrap it.